ckan/controllers/user.py

```diff
@@ -162,10 +166,87 @@ def register(self, data=None, errors=None, error_summary=None):
 
         return self.new(data, errors, error_summary)
 
+    def _generate_internal_logs(self, log_message):
+        '''
+        This helper function is called if sending a new user registration email fails.
+        '''
+        logging.warning("There was an error sending the email. Writing to log file.")
+
+        try:
+            log_path = os.environ.get('REGISTRATION_ERROR_LOG_FILE_PATH')
+            log_file = os.environ.get('REGISTRATION_ERROR_LOG_FILE_NAME')
```

Comment:
`.get()` won't raise an exception if the environment variable doesn't exist. just check that the log_path and log_file aren't None; `if not log_path or not log_file:`